src/common/utility.ts

```diff
@@ -627,6 +627,61 @@ export class Utility {
             validateFunc, Constants.moduleNameDft);
     }
 
+    public static async checkDockerState(outputChannel: vscode.OutputChannel) {
+        let state: DockerState;
+        let errorMsg: any;
+        try {
+            await Executor.executeCMD(outputChannel, "docker", { shell: true }, "version");
+            state = DockerState.Runing;
+        } catch (error) {

```

Comment:
Is error code same between "command not found" and "not running"?

Reply:
Have we checked the error code of the following cases?

Reply:
Yes, they are both 1, so it is useless.

Reply:
For linux , if docker is not installed, it would 127. And I will update to handle this